pandas/tests/series/methods/test_compare.py

```diff
@@ -138,3 +138,17 @@ def test_compare_datetime64_and_string():
     tm.assert_series_equal(result_eq1, expected_eq)
     tm.assert_series_equal(result_eq2, expected_eq)
     tm.assert_series_equal(result_neq, expected_neq)
+
+
+def test_comparison_string_mixed_object():
+    # Issue https://github.com/pandas-dev/pandas/issues/60228
+    pd.options.future.infer_string = True
```

Comment:
You don't need to add this for CI, because we have a separate CI build that enables this option for the full test suite.

Now, this can still be useful to test locally, but the way you can do this is with setting an environment variable (on linux I can do `PANDAS_FUTURE_INFER_STRING=1 pytest ...` to run the test with the option enabled.